src/stories/packages/components/Modal.stories.tsx

```diff
@@ -0,0 +1,123 @@
+import type { Meta, StoryObj } from '@storybook/react';
+import { ButtonPrimary, Modal, ModalProps } from '@terra-ui-packages/components';
+import React, { CSSProperties, useState } from 'react';
+
+const meta: Meta<typeof Modal> = {
+  title: 'Packages/Components/Modal',
+  component: Modal,
+  parameters: {
+    layout: 'centered',
+  },
+  tags: ['autodocs'],
+  argTypes: {
+    cancelText: {
+      control: 'text',
+      description: 'text of Cancel button',
+    },
+    children: {
+      control: 'text',
+      description: 'modal body (string OR components)',
+    },
+    danger: {
+      control: 'boolean',
+      description: 'is the action dangerous?',
```

Comment:
Nit: Would "destructive" more clearly describe the intent here?

```suggestion
      description: 'is the action destructive?',
```